models/user.go

```diff
@@ -275,24 +275,73 @@ func (u *User) ConfirmEmailChange(tx *storage.Connection, status int) error {
 	u.EmailChangeTokenCurrent = ""
 	u.EmailChangeTokenNew = ""
 	u.EmailChangeConfirmStatus = status
-	return tx.UpdateOnly(
+
+	if err := tx.UpdateOnly(
 		u,
 		"email",
 		"email_change",
 		"email_change_token_current",
 		"email_change_token_new",
 		"email_change_confirm_status",
-	)
+	); err != nil {
+		return err
+	}
+
+	identity, err := FindIdentityByIdAndProvider(tx, "email", u.ID.String())
+	if err != nil {
+		if IsNotFoundError(err) {
+			// no email identity, not an error
+			return nil
+		}
+
+		return err
+	}
+
+	identity.IdentityData["email"] = u.EmailChange
+
+	if err := tx.UpdateOnly(u, "identity_data"); err != nil {
```

Comment:
hmm wouldn't this fail? i don't think there is an `identity_data` column on the `auth.users` table?